catalog/transformer/catalog-transformer-mgmp/src/test/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverterTest.java

```diff
@@ -0,0 +1,465 @@
+        System.out.println(xml);
+        try {
+            schema.newValidator()
+                    .validate(new StreamSource(new StringReader(xml)));
+        } catch (SAXException | IOException e) {
+            fail("Generated MGMPv2 Response does not conform to Schema: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testSchemaComplianceWithoutDescription() {
+        metacard.setAttribute(Core.DESCRIPTION, (Serializable) Collections.emptyList());
+        validateXml(generateMgmpXml());
+    }
+
+    @Test
+    public void testSchemaCompliance() {
+        validateXml(generateMgmpXml());
+    }
+
+    @Test
+    public void testSchemaComplianceEmptyPublisher() {
+        metacard.setAttribute(Contact.PUBLISHER_NAME, (Serializable) Collections.emptyList());
+        metacard.setAttribute(Contact.PUBLISHER_PHONE, (Serializable) Collections.emptyList());
+        metacard.setAttribute(Contact.PUBLISHER_ADDRESS, (Serializable) Collections.emptyList());
+        metacard.setAttribute(Contact.PUBLISHER_EMAIL, (Serializable) Collections.emptyList());
+
+        validateXml(generateMgmpXml());
+    }
+
+    @Test
+    public void testSchemaComplianceEmptyPointOfContact() {
+        metacard.setAttribute(Contact.POINT_OF_CONTACT_NAME,
+                (Serializable) Collections.emptyList());
+        metacard.setAttribute(Contact.POINT_OF_CONTACT_PHONE,
+                (Serializable) Collections.emptyList());
+        metacard.setAttribute(Contact.POINT_OF_CONTACT_ADDRESS,
+                (Serializable) Collections.emptyList());
+        metacard.setAttribute(Contact.POINT_OF_CONTACT_EMAIL,
+                (Serializable) Collections.emptyList());
+
+        validateXml(generateMgmpXml());
+    }
+
+    @Test
+    public void testSchemaComplianceEmptyKeywords() {
+        metacard.setAttribute(Topic.KEYWORD, (Serializable) Collections.emptyList());
+
+        validateXml(generateMgmpXml());
+    }
+
+    @Test
+    public void testSchemaComplianceEmptyTopics() {
+        metacard.setAttribute(Topic.CATEGORY, (Serializable) Collections.emptyList());
+
+        validateXml(generateMgmpXml());
+    }
+
+    @Test
+    public void testMultiplePointOfContactNames() {
+        metacard.setAttribute(Contact.POINT_OF_CONTACT_NAME,
+                (Serializable) Arrays.asList("Name1", "Name2"));
+
+        validateXml(generateMgmpXml());
+    }
+
+    /**
+     * Test that the generated xml matches the reference xml.
+     */
+    @Test
+    public void testMarshal() throws IOException, SAXException {
+        String compareString;
+        try (InputStream input = getClass().getResourceAsStream("/xmldiff-reference-mgmp.xml")) {
+            compareString = IOUtils.toString(input);
+        }
+
+        String xml = generateMgmpXml();
+        Diff diff = new Diff(compareString, xml);
+
+        LOGGER.info("diff:\n" + diff.toString());
+        assertThat(diff.identical(), is(true));
+    }
```

Comment:
The team discussed the unit testing at length. We concluded that the current approach was the best to start with. Additional edge can be added as they are discovered. But trying to come up with a meaningful list of cases for initial testing would result in a huge list of tests.